app/javascript/src/NewService/components/FormElements/ServiceManualListItems.jsx

```diff
@@ -1,36 +1,51 @@
-    <li id="service_name_input" className="string required">
-      <Label
-        htmlFor='service_name'
-        label='Name'
-        required
-      />
-      <input maxLength="255" id="service_name" type="text" name="service[name]" autoFocus="autoFocus"/>
-    </li>
-    <li id="service_system_name_input" className="string required">
-      <Label
-        htmlFor='service_system_name'
-        label='System name'
-        required
-      />
-      <input maxLength="255" id="service_system_name" type="text" name="service[system_name]"/>
-      <p className="inline-hints">
-        Only ASCII letters, numbers, dashes and underscores are allowed.
-      </p>
-    </li>
-    <li id="service_description_input" className="text optional">
-      <Label
-        htmlFor='service_description'
-        label='Description'
-      />
-      <textarea rows="3" id="service_description" name="service[description]"></textarea>
-    </li>
-  </React.Fragment>
+type Props = ServiceFormTemplate
 
+const ServiceManualListItems = ({service, errors}: Props) => {
+  const [name, setName] = useState(service.name)
+  const [systemName, setSystemName] = useState(service.system_name)
+  const [description, setDescription] = useState(service.description)
+  const onChange = (fn) => (e) => fn(e.currentTarget.value)
+
+  return (
+    <React.Fragment>
+      { /* $FlowFixMe boolean coercion in string literals */ }
+      <li id="service_name_input" className={`string required ${!!errors.name && 'error'}`}>
+        <Label
+          htmlFor='service_name'
+          label='Name'
+          required
+        />
+        <input onChange={onChange(setName)} value={name} maxLength="255" id="service_name" type="text" name="service[name]" autoFocus="autoFocus"/>
+        { !!errors.name && <p className="inline-errors">{errors.name}</p> }
+      </li>
+      { /* $FlowFixMe boolean coercion in string literals */ }
+      <li id="service_system_name_input" className={`string required ${!!errors.system_name && 'error'}`}>
+        <Label
+          htmlFor='service_system_name'
+          label='System name'
+          required
+        />
+        <input onChange={onChange(setSystemName)} value={systemName} maxLength="255" id="service_system_name" type="text" name="service[system_name]"/>
+        <p className={`${errors.system_name ? 'inline-errors' : 'inline-hints'}`}>
+          Only ASCII letters, numbers, dashes and underscores are allowed.
+        </p>
+      </li>
+      { /* $FlowFixMe boolean coercion in string literals */ }
+      <li id="service_description_input" className={`text optional ${!!errors.description && 'error'}`}>
+        <Label
+          htmlFor='service_description'
+          label='Description'
+        />
+        <textarea onChange={onChange(setDescription)} value={description} rows="3" id="service_description" name="service[description]"></textarea>
+        { !!errors.description && <p className="inline-errors">{errors.description}</p> }
```

Comment:
What are the possible errors for "Description" ?
I've just created a product with empty description.
I think the description is optional, in that case better remove this `p`

Reply:
Maybe when it's too long? Would the server return an error or ..?

Reply:
`Only ASCII letters, numbers, dashes and underscores are allowed`

Reply:
ok, it validates length, maximum 65535... a lot, but still potentially could fail